Source/Lib/Codec/EbThreads.c

```diff
@@ -80,8 +80,8 @@ EB_HANDLE EbCreateThread(
 
         if (ret != 0) {
             if (ret == EPERM) {
-
                 pthread_cancel(*((pthread_t*)threadHandle));
+                pthread_join(*((pthread_t*)threadHandle), NULL);
```

Comment:
```
       int pthread_create(pthread_t *thread, const pthread_attr_t *attr,
                          void *(*start_routine) (void *), void *arg); 
 On success, pthread_create() returns 0; on error, it returns an error number, and the contents of *thread are undefined.
```
from: https://man7.org/linux/man-pages/man3/pthread_create.3.html.
Since *thread Do we even need to call pthread_cancel and pthread_join in this case?

Reply:
In either case the two levels of return code checking can be condensed (i.e. ret != 0) is not needed.

        if (ret != 0) {
            if (ret == EPERM) {

Reply:
> ```
>        int pthread_create(pthread_t *thread, const pthread_attr_t *attr,
>                           void *(*start_routine) (void *), void *arg); 
>  On success, pthread_create() returns 0; on error, it returns an error number, and the contents of *thread are undefined.
> ```
> 
> from: https://man7.org/linux/man-pages/man3/pthread_create.3.html.
> Since *thread is undefined, do we even need to call pthread_cancel and pthread_join in this case?

good question:), actually I don't think it's a good idea to turn set the thread policy to real time, it will drain the CPU. Of course it's good for encoding, and can have a little better encoder performance. But in real use case where you have other components like decoders/demuxers etc, doing so will not end up well